src/EPiServer.SocialAlloy.Web/Social/Repositories/Groups/CommunityMemberRepository.cs

```diff
@@ -100,41 +96,37 @@ public IEnumerable<CommunityMember> Get(CommunityMemberFilter communityMemberFil
             {
                 throw new SocialRepositoryException("Episerver Social failed to process the application request.", ex);
             }
-
-            return returnedMembers;
         }
 
         /// <summary>
-        /// Build the appropriate CompositeCriteria based the provided CommunityMemberFilter.
-        /// The member filter will either contain a group id or a logged in user id. If neitheris provided an exception is thrown.
+        /// Build the appropriate criteria based the provided CommunityMemberFilter.
+        /// The member filter will either contain a group id or a logged in user id. 
+        /// If neither is provided an exception is thrown.
         /// </summary>
         /// <param name="communityMemberFilter">The provided member filter</param>
         /// <returns>A composite criteria of type MemberFilter and MemberExtensionData</returns>
-        private CompositeCriteria<MemberFilter, MemberExtensionData> BuildCriteria(CommunityMemberFilter communityMemberFilter)
+        private Criteria BuildCriteria(CommunityMemberFilter communityMemberFilter)
         {
-            var pageInfo = new PageInfo { PageSize = communityMemberFilter.PageSize };
-            var orderBy = new List<SortInfo> { new SortInfo(MemberSortFields.Id, false) };
-            var compositeCriteria = new CompositeCriteria<MemberFilter, MemberExtensionData>()
+            FilterExpression filter = null;
+            if (!string.IsNullOrEmpty(communityMemberFilter.CommunityId))
             {
-                PageInfo = pageInfo,
-                OrderBy = orderBy
-            };
-
-            if (!string.IsNullOrEmpty(communityMemberFilter.CommunityId) && (string.IsNullOrEmpty(communityMemberFilter.UserId)))
-            {
-                compositeCriteria.Filter = new MemberFilter { Group = GroupId.Create(communityMemberFilter.CommunityId) };
-
+                filter = this.memberFilters.Group.EqualTo(GroupId.Create(communityMemberFilter.CommunityId));
             }
-            else if ((!string.IsNullOrEmpty(communityMemberFilter.UserId) && (string.IsNullOrEmpty(communityMemberFilter.CommunityId))))
+            else if (!string.IsNullOrEmpty(communityMemberFilter.UserId))
             {
-                compositeCriteria.Filter = new MemberFilter { User = Reference.Create(communityMemberFilter.UserId) };
+                filter = this.memberFilters.User.EqualTo(Reference.Create(communityMemberFilter.UserId));
             }
             else
             {
                 throw new SocialException("This implementation of a CommunityMemberFilter should only contain either a CommunityId or a UserReference.");
             }
 
-            return compositeCriteria;
+            return new Criteria
+            {
+                PageInfo = new PageInfo { PageSize = communityMemberFilter.PageSize },
+                OrderBy = new List<SortInfo> { new SortInfo(MemberSortFields.Id, false) },
+                Filter = filter
```

Comment:
`.Is<T>()` will filter by .NET type rather than type ID. `Is("...ID..")` will filter by type ID. But in this example, yes, they will produce the same result.